packages/arb-token-bridge-ui/src/components/Widget/WidgetConnectWalletButton.tsx

```diff
@@ -0,0 +1,42 @@
+import { useCallback } from 'react'
+import { useArbQueryParams } from '../../hooks/useArbQueryParams'
+import { ConnectWalletButton } from '../TransferPanel/ConnectWalletButton'
+
+/**
+ * A wrapper around ConnectWalletButton that handles mobile browser restrictions in embed mode.
+ *
+ * Mobile browsers block iframes from accessing cookies and web storage, which breaks wallet connections.
+ * When detected, users are redirected to bridge.arbitrum.io with their query parameters preserved.
+ */
+
+const isMobileBrowser = () => {
+  if (typeof window === 'undefined') return false
+
+  const userAgent = window.navigator.userAgent.toLowerCase()
+  return /android|webos|iphone|ipad|ipod|blackberry|iemobile|opera mini/i.test(
+    userAgent
+  )
+}
+
+export function WidgetConnectWalletButton() {
+  const [{ embedMode }] = useArbQueryParams()
+  const isMobile = isMobileBrowser()
+
+  const handleClick = useCallback(() => {
+    if (embedMode && isMobile && window.top) {
```

Comment:
we're checking for that condition both here and on line 39, we can remove one of those check

Reply:
Added in 2 places to avoid type issues